src/Modules/OrchardCore.Commerce/Services/CardPaymentService.cs

```diff
@@ -0,0 +1,157 @@
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace OrchardCore.Commerce.Services;
+
+public class CardPaymentService : ICardPaymentService
+{
+    private readonly IShoppingCartPersistence _shoppingCartPersistence;
+    private readonly IPriceService _priceService;
+    private readonly PaymentIntentService _paymentIntentService;
+    private readonly IPriceSelectionStrategy _priceSelectionStrategy;
+    private readonly IContentManager _contentManager;
+    private readonly ISiteService _siteService;
+    private readonly IDataProtectionProvider _dataProtectionProvider;
+    private readonly ILogger<CardPaymentService> _logger;
+
+    public CardPaymentService(
+        IShoppingCartPersistence shoppingCartPersistence,
+        IPriceService priceService,
+        IPriceSelectionStrategy priceSelectionStrategy,
+        IContentManager contentManager,
+        ISiteService siteService,
+        IDataProtectionProvider dataProtectionProvider,
+        ILogger<CardPaymentService> logger)
+    {
+        _paymentIntentService = new PaymentIntentService();
+        _shoppingCartPersistence = shoppingCartPersistence;
+        _priceService = priceService;
+        _priceSelectionStrategy = priceSelectionStrategy;
+        _contentManager = contentManager;
+        _siteService = siteService;
+        _dataProtectionProvider = dataProtectionProvider;
+        _logger = logger;
+    }
+
+    public async Task<PaymentIntent> CreatePaymentAsync(ConfirmPaymentRequest request)
+    {
+        var currentShoppingCart = await _shoppingCartPersistence.RetrieveAsync();
+        var totals = await currentShoppingCart.CalculateTotalsAsync(_priceService, _priceSelectionStrategy);
+
+        // Same here as on the checkout page: Later we have to figure out what to do if there are multiple
+        // totals i.e., multiple currencies.
+        var defaultTotal = totals.FirstOrDefault();
+
+        var paymentIntent = new PaymentIntent();
+        var requestOptions =
+            new RequestOptions
+            {
+                ApiKey = (await _siteService.GetSiteSettingsAsync())
+                    .As<StripeApiSettings>()
+                    .SecretKey
+                    .DecryptStripeApiKey(_dataProtectionProvider, _logger),
+            };
+
+        if (request.PaymentMethodId != null)
+        {
+            var paymentIntentOptions = new PaymentIntentCreateOptions
+            {
+                // We need to remove the decimal points and convert the value (decimal) to long.
+                // https://stripe.com/docs/currencies#zero-decimal
+                Amount = long
+                .Parse(
+                    string.Join(
+                        string.Empty,
+                        defaultTotal.ToString().Where(char.IsDigit)),
+                    CultureInfo.InvariantCulture),
+                Currency = defaultTotal.Currency.CurrencyIsoCode,
+                Description = "Orchard Commerce Test Stripe Card Payment",
```

Comment:
I don't think this should be hard coded?

Reply:
It shouldn't. What should I do? Maybe just remove it for now?

Reply:
Can't it be a property in `ConfirmPaymentRequest`?

Reply:
It can be, but what value would we give it, and from where?

Reply:
Dunno, that's why I'm asking. :)
Or for the time being it could be just `T["Payment for {0}", siteSettings.SiteName].Value`.

Reply:
Good idea!